src/modules/esl-event-listener/core/targets/intersection.event.ts

```diff
@@ -5,7 +5,12 @@ import {overrideEvent} from '../../../esl-utils/dom/events/misc';
  * Based on the {@link IntersectionObserverEntry} object.
  */
 export class ESLIntersectionEvent extends Event implements IntersectionObserverEntry {
-  public static readonly type = 'intersects';
+  public static readonly OUT_TYPE = 'intersects:out';
+  public static readonly IN_TYPE = 'intersects:in';
+  public static readonly BASE_TYPE = 'intersects';
```

Comment:
Oh we are having a mess in event types constants:
 - `ESLIntersectionEvent.type` (and I think the most Events in `esl-event-listener` module)
 - `ESLCarouselSlideEvent.BEFORE` and `ESLCarouselSlideEvent.BEFORE`  (w/o postfix)
 
 In terms of what will be the ideal, I'd say:
  1. I'd not expect any other contents than the constants of the event name; I'd vote for names without postfixes/prefixes (moreover, for us, it's an extra bytes in a bundle). So for that one `IN`, `OUT`, `TYPE` (or `BASE`)
  2. Regarding the case, you are more right to choose the upper case as we are dealing with constants.
  The most complex one:
  3. The name for the most general event ... I'd actually vote for just `TYPE`, but not sure at all
  
  The only thing that I'm sure with: in case we are replacing the `type` we should deprecate it.
  
  @exadel-inc/esl-core-maintainers, you probably have some preferences/ideas regarding naming for event name constants. From my side, it's just thoughts; the only thing we need for sure is a consistent approach.